### PR TITLE
add otel-dev sample otlp app to argocd

### PR DESCRIPTION
**Pull request by sallyom**
/cc @HumairAK 
let's see if this works

### CODE REVIEW

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/hello-opentelemetry/hello-opentelemetry.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: smaug
+    namespace: otel-dev
+  source:
+    repoURL: 'https://github.com/sallyom/hello-opentelemetry'
+    path: opentelemetry-sample-app/charts/hello-opentelemetry
+    targetRevision: HEAD
+    helm:
+      valueFiles:
+        values.yaml
+
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/applications/envs/moc/smaug/hello-opentelemetry/hello-opentelemetry.yaml:18]: @sallyom can you remove this newline?

File: argocd/overlays/moc-infra/projects/otel-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: hello-opentelemetry
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/projects/otel-dev.yaml:4]: did you mean to name this argocd project `octo-otel-dev`?

Comment (sallyom) [reply argocd/overlays/moc-infra/projects/otel-dev.yaml:4]: yes - I was going to name it  hello-opentelemetry project but was confused about whether it should match what already exists. What's the meaning of the `otel-dev-*` here? Will other namespaces/projects be created? 
```yaml
spec:
  destinations:
    - namespace: 'otel-dev-*'
      server: 'https://api.smaug.na.operate-first.cloud:6443
```

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/hello-opentelemetry/hello-opentelemetry.yaml (hunk 1 above)
Comment (sallyom) [reply argocd/overlays/moc-infra/applications/envs/moc/smaug/hello-opentelemetry/hello-opentelemetry.yaml]: updated

File: argocd/overlays/moc-infra/projects/otel-dev.yaml (hunk 2 above)
Comment (sallyom) [reply argocd/overlays/moc-infra/projects/otel-dev.yaml]: for now I changed it to `octo-otel-dev` that works.
I installed the helm charts already in otel-dev ns - that's what is running in there now

File: argocd/overlays/moc-infra/projects/otel-dev.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/projects/otel-dev.yaml:2]: @sallyom can you also include `otel-dev.yaml` project [here](https://github.com/operate-first/apps/blob/master/argocd/overlays/moc-infra/projects/kustomization.yaml)?

Comment (sallyom) [reply argocd/overlays/moc-infra/projects/otel-dev.yaml:2]: done!

**Pull request merged**